src/test/java/com/github/tomakehurst/wiremock/client/WireMockClientAcceptanceTest.java

```diff
@@ -70,4 +73,60 @@ public void buildsMappingWithUrlOnyRequestAndResponseWithJsonBodyWithDiacriticSi
     assertThat(
         testClient.get("/my/new/resource").content(), is("{\"address\":\"Puerto Banús, Málaga\"}"));
   }
+
+  @Test
+  void testGetOrHeadRequestWhenGetMatchesShouldReturnAResponseBody() {
+    String path = "/get-or-head-test";
+    WireMock wireMock = WireMock.create().port(wireMockServer.port()).build();
+    wireMock.register(getOrHead(urlEqualTo(path)).willReturn(okJson("{\"key\": \"value\"}")));
+
+    WireMockResponse response = testClient.get(path);
+
+    assertThat(response.statusCode(), is(200));
+    assertThat(response.firstHeader("Content-Type"), is("application/json"));
+    assertThat(response.content(), not(emptyOrNullString()));
+  }
+
+  @Test
+  void testGetOrHeadRequestWhenHeadMatchesShouldNotReturnAResponseBody() {
+    String path = "/get-or-head-test";
+    WireMock wireMock = WireMock.create().port(wireMockServer.port()).build();
+
+    wireMock.register(
+        getOrHead(urlEqualTo(path))
+            .willReturn(
+                okJson("{\"key\": \"value\"}").withHeader("Content-Type", "application/json")));
+    WireMockResponse response = testClient.head(path);
+
+    assertThat(response.statusCode(), is(200));
+    assertThat(response.firstHeader("Content-Type"), is("application/json"));
+    assertThat(response.content(), is(emptyOrNullString()));
```

Comment:
Would be great to see an assertion about `Content-Length` matching the length of the response body.